src/python/pants/engine/fs.py

```diff
@@ -73,6 +73,14 @@ class FileEntry:
     is_executable: bool = False
 
 
+@dataclass(frozen=True)
+class LinkEntry:
```

Comment:
This should probably be called `SymlinkEntry` for consistency with usages of "symlink" elsewhere (including in the REAPI where there is a `SymlinkNode` to represent them).

Reply:
Meanwhile, we use `Link` throughout `fs.rs`.